tripy/tripy/frontend/trace/ops/pad.py

```diff
@@ -0,0 +1,145 @@
+from tripy.frontend.trace.ops.base import BaseTraceOp
+from tripy.common.exception import raise_error
+
+
+@dataclass(repr=False)
+class Pad(BaseTraceOp):
+
+    padding_value: Union[int, float]
+
+    infer_shape_output_idxs = op_utils.ShapeOutputIdxPolicies.never_return_shape
+
+    def infer_dtypes(self):
+        self.outputs[0].dtype = self.inputs[0].dtype
+
+    def infer_rank(self):
+        self.outputs[0].rank = self.inputs[0].rank
+
+    def to_flat_ir(self, inputs, outputs):
+        from tripy.common.datatype import int32
+        from tripy.flat_ir.ops import ConstantOp, DynamicPadOp
+        from tripy.flat_ir.tensor import FlatIRTensor
+
+        pad_val_tensor = FlatIRTensor.build(
+            shape=(),
+            rank=0,
+            dtype=outputs[0].dtype,
+            device=outputs[0].device,
+            reason_details=[f"create the constant value tensor (containing {self.padding_value}) for a pad operation"],
+        )
+        ConstantOp.build([], [pad_val_tensor], data=self.padding_value)
+
+        # interior_padding is not supported
+        # create the default value
+        pad_size_shape = (inputs[0].rank,)
+        interior_pad_tensor = FlatIRTensor.build(
+            shape=pad_size_shape,
+            rank=1,
+            dtype=int32,
+            device=outputs[0].device,
+            reason_details=[f"create the default value for interior_padding argument."],
+        )
+        ConstantOp.build([], [interior_pad_tensor], data=[0] * inputs[0].rank)
+
+        # [operand, pad_val, low, high, interior]
+        inputs.insert(1, pad_val_tensor)
+        inputs.append(interior_pad_tensor)
+        # set padding size tensors' shape
+        # because stablehlo requires static shapes
+        inputs[2].shape = pad_size_shape
+        inputs[3].shape = pad_size_shape
+        DynamicPadOp.build(inputs, outputs)
+
+
+def _convert_pad_sizes(padding_sizes):
+    from tripy.common.datatype import int32
+    from tripy.frontend.tensor import Tensor
+    from tripy.frontend.trace.ops.concatenate import concatenate
+    from tripy.frontend.trace.ops.unsqueeze import unsqueeze
+
+    if not any(isinstance(s, Tensor) for s in padding_sizes):
+        return Tensor(padding_sizes, dtype=int32)
+
+    sizes_1d = []
+    for size in padding_sizes:
+        if isinstance(size, Tensor):
+            assert size.rank == 0, f"Size expected to be of rank 0, got {size.rank}."
+            sizes_1d.append(unsqueeze(size, 0))
+        else:
+            sizes_1d.append(Tensor([size], dtype=int32))
+    return concatenate(sizes_1d, 0)
+
+
+@export.public_api(document_under="operations/functions")
+@constraints.dtype_info(
+    dtype_variables={
+        "T1": ["float32", "float16", "bool", "int32"],
+    },
+    dtype_constraints={"input": "T1", constraints.RETURN_VALUE: "T1"},
+)
+def pad(
+    input: "tripy.Tensor",
+    padding_sizes: Sequence[Tuple[Union[int, "tripy.ShapeScalar"]]],
+    padding_value: Union[int, float] = 0,
```

Comment:
Can we use `pad` and `value` for these? Also, should we add a `mode` argument so we can eventually support things like reflection padding?

Reply:
Do you mean changing the argument name to `pad` and `value`? The function name is `pad`, might be a bit confusing.

Yes we should have a `mode` argument, but stablehlo dynamic_pad op does not support it. We will need tensorrt dialect for that, could discuss with Chris next time.

Reply:
Yeah, `pad` and `value` is what torch uses. We can add a `mode` argument that only supports `constant` for now maybe? Then we won't need API changes later.

Reply:
Updated the API.